src/injected/content/inject.js

```diff
@@ -40,14 +43,29 @@ export function triageScripts(data) {
       return false;
     });
     let support;
-    data.scripts.forEach((script) => {
-      let injectInto = script.custom.injectInto || script.meta.injectInto || data.injectInto;
-      if (injectInto === INJECT_AUTO) {
+    const injectChecking = {
+      [INJECT_PAGE]: () => {
         if (!support) support = { injectable: checkInjectable() };
-        injectInto = support.injectable ? INJECT_PAGE : INJECT_CONTENT;
-      }
-      const list = scriptLists[injectInto];
-      if (list) list.push(script);
+        return support.injectable;
+      },
+      [INJECT_WRAP]: () => isFirefox,
+      [INJECT_CONTENT]: () => true,
+    };
+    const internalModes = {
+      // `wrap` mode works only for Firefox, fallback to `content` if not supported
+      [INJECT_WRAP]: [INJECT_WRAP, INJECT_CONTENT],
+      // `auto` tries to provide `window` from the real page as `unsafeWindow`
+      [INJECT_AUTO]: [INJECT_PAGE, INJECT_WRAP, INJECT_CONTENT],
+    };
+    data.scripts.forEach((script) => {
+      const injectInto = script.custom.injectInto || script.meta.injectInto || data.injectInto;
+      const internalInjectInto = internalModes[injectInto] || [injectInto];
+      const availableInjectInto = internalInjectInto.find(key => {
+        const check = injectChecking[key];
+        return check && check();
+      });
+      const list = scriptLists[availableInjectInto];
+      if (list) list.push({ script, injectInto: availableInjectInto });
```

Comment:
`list?.push({ script, injectInto: availableInjectInto });`

And to fix the [false-positive error from ESLint](https://github.com/eslint/eslint/issues/11045#issuecomment-436685184) the following should be added to `.eslintrc`:

```js
  rules: {
    'prefer-object-spread': 'off',
    // workaround for the false-positive on optional?.chainingCall() statement
    // https://github.com/eslint/eslint/issues/11045#issuecomment-436685184
    'no-unused-expressions': 'off',
    'babel/no-unused-expressions': 'error',
  },
```